src/Components/Components/src/Routing/QueryParameterValueSupplier.cs

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Microsoft.AspNetCore.Components.Reflection;
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.AspNetCore.Internal;
+using static Microsoft.AspNetCore.Internal.LinkerFlags;
+
+namespace Microsoft.AspNetCore.Components.Routing
+{
+    internal sealed class QueryParameterValueSupplier
+    {
+        private static readonly Dictionary<Type, QueryParameterValueSupplier?> _cacheByType = new();
```

Comment:
We'll need to clear this on a hot reload event. 

For reference:

* https://github.com/dotnet/aspnetcore/blob/main/src/Components/Components/src/Reflection/ComponentProperties.cs#L23 
* https://github.com/dotnet/aspnetcore/blob/main/src/Components/Components/src/RenderTree/Renderer.cs#L131

Reply:
Very good point - thanks!

Reply:
Done